src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -1005,45 +1015,61 @@ private static BinderCheckTypeResult CheckEqualArgTypesCore(IErrorContainer erro
                 return new BinderCheckTypeResult();
             }
 
-            // Special case for comparing option set values, it should produce an error when the base option sets are different
-            if (typeLeft.Kind == DKind.OptionSetValue && typeRight.Kind == DKind.OptionSetValue && !typeLeft.Accepts(typeRight, exact: true, useLegacyDateTimeAccepts: false, usePowerFxV1CompatibilityRules: usePowerFxV1CompatibilityRules))
+            // Special case for comparing option set values
+            if (typeLeft.Kind == DKind.OptionSetValue || typeRight.Kind == DKind.OptionSetValue)
             {
+                // Comparing values from two different option set values is not supported, with or without V1 compat rules
```

Comment:
>// Comparing values from two different option set values is not supported, with or without V1 compat rules [](http://example.com/codeflow?start=16&length=106)

Today you can compare two different enums (e.g., `ImageRotation.None = GridStyle.None`, `JSONFormat.IgnoreUnsupportedTypes = TraceOptions.IgnoreUnsupportedTypes`). Will this break?

Reply:
That comment was a tad overzealous.  Without StronglyTypedBuiltinEnums, option sets are converted to their backing kinds before getting this far.  I update the comments.

Moving forward, yes we are already blocking this in main with the original Strongly Typed implementation, which I think is correct.  There are lots of tests added to check this, 
https://github.com/microsoft/Power-Fx/blob/0932cfac9d2735b87705db1b4e3d6749a0b06861/src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StronglyTypedEnum_BuiltInEnums.txt#L219
as well as checking that we don't break this for Canvas
https://github.com/microsoft/Power-Fx/blob/0932cfac9d2735b87705db1b4e3d6749a0b06861/src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/StronglyTypedEnum_BuiltInEnums_PreV1.txt#L221